### PR TITLE
Added transitive closure to change-impact based on dependancy graph

### PR DESCRIPTION
**Pull request by AnnaTrost**

### CODE REVIEW

File: src/goto-diff/goto_diff_parse_options.h [hunk 1]
```diff
@@ -23,7 +23,8 @@ class optionst;
   "(json-ui)" \
   "(show-goto-functions)" \
   "(verbosity):(version)" \
-  "u(unified)(change-impact)"
+  "u(unified)(change-impact)(forward-impact)(backward-impact)" \
+  "(demo-output)"
```
Comment (tautschnig) [inline src/goto-diff/goto_diff_parse_options.h]: I'm not sure "demo-output" is a good choice: command-line options are for eternity, make sure they are almost self-explanatory.

File: src/goto-diff/goto_diff_parse_options.cpp [hunk 2]
```diff
@@ -335,9 +335,13 @@ int goto_diff_parse_optionst::doit()
     return 0;
   }
 
-  if(cmdline.isset("change-impact"))
+  if(cmdline.isset("change-impact") || cmdline.isset("forward-impact") || cmdline.isset("backward-impact"))
```
Comment (tautschnig) [inline src/goto-diff/goto_diff_parse_options.cpp]: Apologies in advance for nit-picking, but to keep the code base clean: lines should, where reasonably possible, be no longer than 70 characters.

File: src/goto-diff/goto_diff_parse_options.cpp [hunk 3]
```diff
@@ -335,9 +335,13 @@ int goto_diff_parse_optionst::doit()
...
   {
-    change_impact(goto_model1, goto_model2);
+	impact_modet impact_mode =
```
Comment (tautschnig) [inline src/goto-diff/goto_diff_parse_options.cpp]: Please check the indent.

**Comment by AnnaTrost** [comment]
I did not realize that commits post pull request will be included, I only intended to add the first commit to this branch.
I will pull again when I'll finish working on the demo and fixing some more bugs I found.
Apologies,
Anna

**Pull request closed without merging**